### PR TITLE
migrated input events

### PR DESCRIPTION
**Pull request by amazingalek**
Some comments:
* Input events are now global and shared. Side-effects?
* Added some `todo` comments where I didn't know what to do.
* ModEvents are now passed around in menus *a lot*, might wanna consider a singleton to access OWML's event object.

@TAImatem feel free to take over this branch/PR (or start from scratch if this is crap), you know the input system much better than me.

### CODE REVIEW

**Comment by TAImatem** [comment]
Input events should work a bit better when global actually

**Comment by TAImatem** [comment]
I suppose input can self-instance very easily, might as well do a static Instance thing

File: OWML.ModHelper.Menus/ModMenuWithSelectables.cs [hunk 1]
```diff
@@ -15,21 +14,23 @@ public class ModMenuWithSelectables : ModPopupMenu, IModMenuWithSelectables
...
-            CommandListener.AddToListener(InputLibrary.setDefaults);
-            CommandListener.OnNewlyReleased += OnButton;
-            listenerObject.SetActive(false);
+            _events.Input.AddToListener(InputLibrary.confirm);
+            _events.Input.AddToListener(InputLibrary.enter2);//keypad's Enter
+            _events.Input.AddToListener(InputLibrary.cancel);
+            _events.Input.AddToListener(InputLibrary.escape);
+            _events.Input.AddToListener(InputLibrary.setDefaults);
+            _events.Input.OnNewlyReleased += OnButton;
+            //listenerObject.SetActive(false); todo hmmm? :)
```
Comment (amazingalek) [inline OWML.ModHelper.Menus/ModMenuWithSelectables.cs:33]: @TAImatem what should we do here?

File: OWML.ModHelper.Menus/ModMenuWithSelectables.cs [hunk 2]
```diff
@@ -151,13 +149,13 @@ protected virtual void AddSelectable(Selectable selectable)
...
-            CommandListener.gameObject.SetActive(true);
-            CommandListener.BlockNextRelease();
+            //CommandListener.gameObject.SetActive(true); // todo hmmmm
+            _events.Input.BlockNextRelease();
         }
 
         protected virtual void OnDeactivateMenu()
         {
-            CommandListener.gameObject.SetActive(false);
+            //CommandListener.gameObject.SetActive(false); // todo hmmmm
```
Comment (amazingalek) [inline OWML.ModHelper.Menus/ModMenuWithSelectables.cs:158]: and here

**Comment by amazingalek** [comment]
> I suppose InputEvents can self-instantiate very easily, might as well do a static Instance thing
> same thing for UnityEvents actually

I'll experiment with static events, but what do you mean with self-instantiate?

File: OWML.ModHelper.Menus/ModMenuWithSelectables.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.ModHelper.Menus/ModMenuWithSelectables.cs:33]: just change how OnNewlyReleased works I suppose

File: OWML.ModHelper.Menus/ModMenuWithSelectables.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.ModHelper.Menus/ModMenuWithSelectables.cs:158]: ooh. BlockNextRelease probably have to be scrapped now

**Comment by TAImatem** [comment]
Look at how OW does Instance in many classes
that is if when instance is not yet made at the moment of call it just creates a new instance and saves it

**Pull request closed without merging**